### PR TITLE
tidb: remove tidb dependencies.

### PR DESCRIPTION
**Pull request by coocood**
Define multiple main packages for different storage engine, remove `tidb` dependencies.

### CODE REVIEW

File: tidb-server/boltdb/main.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (zimulala) [inline tidb-server/boltdb/main.go]: 2015 -> 2016 ?

File: tidb.go [hunk 2]
```diff
@@ -38,23 +37,17 @@ import (
...
-	"github.com/pingcap/tidb/store/localstore/boltdb"
 	"github.com/pingcap/tidb/store/localstore/engine"
 	"github.com/pingcap/tidb/store/localstore/goleveldb"
 	"github.com/pingcap/tidb/util/types"
 )
 
 // Engine prefix name
 const (
-	EngineGoLevelDBMemory     = "memory://"
-	EngineGoLevelDBPersistent = "goleveldb://"
```
Comment (zimulala) [inline tidb.go]: Why remove it ?

Comment (coocood) [inline tidb.go]: Not used in test files

**Comment by ngaut** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**